tools/java/org/unicode/cldr/test/CheckLogicalGroupings.java

```diff
@@ -205,4 +244,108 @@ public CheckCLDR handleCheck(String path, String fullPath, String value, Options
         }
         return this;
     }
+
+    static final Transliterator SHOW_INVISIBLES = Transliterator.createFromRules(
+        "show",
+        "([[:C:][:Z:][:whitespace:][:Default_Ignorable_Code_Point:]-[\\u0020]]) > &hex/perl($1);",
+        Transliterator.FORWARD);
+
+    public static String showInvisibles(String value) {
+        return SHOW_INVISIBLES.transform(value);
+    }
+
+    public static String showInvisibles(int codePoint) {
+        return showInvisibles(With.fromCodePoint(codePoint));
+    }
+
+    public static String showInvisibles(Multiset<?> codePointCounts) {
+        StringBuilder b = new StringBuilder().append('{');
+        for (Entry<?> entry : codePointCounts.entrySet()) {
+            if (b.length() > 1) {
+                b.append(", ");
+            }
+            Object element = entry.getElement();
+            b.append(element instanceof Integer ? showInvisibles((int) element) : showInvisibles(element.toString()));
+            if (entry.getCount() > 1) {
+                b.append('⨱').append(entry.getCount());
+            }
+        }
+        return b.append('}').toString();
+    }
+
+    private int getMaxDistance(String path, String value, Set<String> paths, Multiset<String> values) {
+        values.clear();
+        final CLDRFile cldrFileToCheck = getCldrFileToCheck();
+        Set<Fingerprint> fingerprints = new HashSet<>();
+        for (String path1 : paths) {
+            final String pathValue = cleanSpaces(path.contentEquals(path1) ? value : cldrFileToCheck.getWinningValue(path1));
+            values.add(pathValue);
+            fingerprints.add(Fingerprint.make(pathValue));
+        }
+        return Fingerprint.maxDistanceBetween(fingerprints);
+    }
+
+    private static final UnicodeMap<String> OTHER_SPACES = new UnicodeMap<String>().putAll(new UnicodeSet("[[:Z:][:S:][:whitespace:]]"), " ").freeze();
+    public static String cleanSpaces(String pathValue) {
+        return OTHER_SPACES.transform(pathValue);
+    }
+
+    // Later
+    private static ConcurrentHashMap<String, Fingerprint> FINGERPRINT_CACHE = new ConcurrentHashMap<>();
```

Comment:
Fingerprint is a measure of a value's characters. Currently just a Multiset with counts for characters; later it could be more sophisticated.